ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/execution/PullQueryExecutorMetrics.java

```diff
@@ -0,0 +1,240 @@
+    this.errorRateSensor = configureErrorRateSensor();
+    this.customMetricsTags = Objects.requireNonNull(customMetricsTags, "customMetricsTags");
+    this.ksqlServiceId = ReservedInternalTopics.KSQL_INTERNAL_TOPIC_PREFIX
+        + ksqlServiceId;
+  }
+
+  @Override
+  public void close() {
+    sensors.forEach(sensor -> metrics.removeSensor(sensor.name()));
+  }
+
+  public void recordLocalRequests(final double value) {
+    this.localRequestsSensor.record(value);
+  }
+
+  public void recordRemoteRequests(final double value) {
+    this.remoteRequestsSensor.record(value);
+  }
+
+  public void recordRate(final double value) {
+    this.requestRateSensor.record(value);
+  }
+
+  public void recordLatency(final double value) {
+    this.latencySensor.record(value);
+  }
+
+  public void recordErrorRate(final double value) {
+    this.errorRateSensor.record(value);
+  }
+
+  private Sensor configureLocalRequestsSensor() {
+    final Sensor sensor = metrics.sensor(
+        PULL_QUERY_METRIC_GROUP + "-" + PULL_REQUESTS + "-local");
+    sensor.add(
+        metrics.metricName(
+            PULL_REQUESTS + "-local-count",
+            ksqlServiceId + PULL_QUERY_METRIC_GROUP,
+            "Count of local pull query requests",
+            customMetricsTags
+        ),
+        new WindowedCount()
+    );
+    sensors.add(sensor);
+    return sensor;
+  }
+
+  private Sensor configureRemoteRequestsSensor() {
+    final Sensor sensor = metrics.sensor(
+        PULL_QUERY_METRIC_GROUP + "-" + PULL_REQUESTS + "-remote");
+    sensor.add(
+        metrics.metricName(
+            PULL_REQUESTS + "-remote-count",
+            ksqlServiceId + PULL_QUERY_METRIC_GROUP,
+            "Count of remote pull query requests",
+            customMetricsTags
+        ),
+        new WindowedCount()
+    );
+    sensors.add(sensor);
+    return sensor;
+  }
+
+  private Sensor configureRateSensor() {
+    final Sensor sensor = metrics.sensor(
+        PULL_QUERY_METRIC_GROUP + "-" + PULL_REQUESTS + "-rate");
+    sensor.add(
+        metrics.metricName(
+            PULL_REQUESTS + "-rate",
+            ksqlServiceId + PULL_QUERY_METRIC_GROUP,
+            "Rate of pull query requests",
+            customMetricsTags
+        ),
+        new Rate()
+    );
+    sensors.add(sensor);
+    return sensor;
+  }
+
+  private Sensor configureErrorRateSensor() {
+    final Sensor sensor = metrics.sensor(
+        PULL_QUERY_METRIC_GROUP + "-" + PULL_REQUESTS + "-error-rate");
+    sensor.add(
+        metrics.metricName(
+            PULL_REQUESTS + "-error-rate",
+            ksqlServiceId + PULL_QUERY_METRIC_GROUP,
+            "Rate of erroneous pull query requests",
+            customMetricsTags
+        ),
+        new Rate()
+    );
+    sensor.add(
+        metrics.metricName(
+            PULL_REQUESTS + "-error-total",
+            ksqlServiceId + PULL_QUERY_METRIC_GROUP,
+            "Total number of erroneous pull query requests",
```

Comment:
do we need a count and a rate? is n't rate sufficient?  (same question for request rate and request count?)..

Reply:
I think its more informative to have a total count and rate. `wrk` reports both as well